src/DGtal/topology/doc/moduleCubicalComplex.dox

```diff
@@ -0,0 +1,328 @@
+space to be valid. You could thus use operations defined in
+DGtal::functions::setops (see \ref moduleSetFunctions) to perform set
+operations and relations on complexes. However, to use the specificity
+of the templated parameter CellContainer of CubicalComplex, it is
+better to use operations and relations defined in namespace
+DGtal::function, for instance as follows:
+
+\code
+using namespace DGtal::functions;
+CC X( K );            // creates the cubical complex
+CC Xc  = ~X;          // closing of X
+CC Xco = *Xc;         // opening of Xc (interior of X).
+CC Xbd = Xc - Xco;    // boundary of X.
+CC Xbd2= X.boundary();// same result for pure complex. 
+\endcode
+
+You have the following operations:
+- functions::operator|  : union of two complexes.
+- functions::operator|= : assignment union with another complex.
+- functions::operator&  : intersection of two complexes.
+- functions::operator&= : assignment intersection with another complex.
+- functions::operator-  : set difference of two complexes.
+- functions::operator-= : assignment set difference with another complex.
+- functions::operator^  : symmetric set difference of two complexes.
+- functions::operator^= : assignment symmetric set difference with another complex.
+- functions::operator~  : closing of a complex (smallest closed complex containg input complex).
+- functions::operator*  : opening of a complex (greatest open complex contained in input complex).
+
+And the following relations:
+- functions::operator<=  : inclusion of two complexes (subset).
+- functions::operator>=  : inclusion of two complexes (supset).
+- functions::operator==  : equality of two complexes (both subset and supset).
+- functions::operator!=  : non-equality of two complexes (equal is false).
+
+All these operations are at most \b linear in the sum of the number of
+cells of input complexes, but they are generally exponential in the
+dimension of complexes.
+
+@section dgtal_ccomplex_sec7 Topological operations: closing, opening, collapsing a complex
+
+You may \b close a complex (CubicalComplex::close), i.e. adding cells
+such that the faces of every cell belongs to the complex. The complex
+is then said to be closed. The updated complex corresponds to the
+smallest closed complex containing it.
+
+You may \b open a complex (CubicalComplex::open), i.e. removing cells
+such that the co-faces of every cell belongs to the complex. The complex
+is then said to be open. The updated complex corresponds to the
+greatest open complex contained in it.
+
+A more difficult topological operation is the \b collapse
+(functions::collapse). A collapse transforms a complex by removing
+cells, but at the end the complex has the same homotopy type as the
+initial complex. The topology is preserved since only free pairs are
+removed. The following snippet shows how to collapse complex X, while
+keeping two cells \a fixed.
+
+@snippet topology/cubical-complex-illustrations.cpp cubical-complex-illustrations-collapse
```

Comment:
ok my bad, there is something line 272. maybe you could detail a little bit this because it is the first usage of a data (flag here) associated with a cell